tslint.json

```diff
@@ -27,6 +27,16 @@
 		"readonly-array": true,
 		"readonly-keyword": [true, "ignore-class"],
 		"strict-boolean-expressions": false,
-		"strict-type-predicates": false
+		"strict-type-predicates": false,
+		// Better to have functionally similar items be close together.
+		"ordered-imports": false,
+		// We need private class properties to start with an underscore.
+		"variable-name": [true, "allow-leading-underscore"],
+		// Better to have functionally similar items be close together.
+		"member-ordering": false,
+		// This rule can sometimes encourage bad practices; better leave this to the PR process.
+		"prefer-function-over-method": false,
```

Comment:
There are some cases:
- Abstract methods: Maybe the method just throws an error (so it doesn't use `this`); if it's designed to be overridden.
- The method is not yet fully implemented; it doesn't use `this` right now, but it will soon, but we want to commit the code anyway.
- One method which doesn't use `this` is functionally related to another method which does use `this`. For example `myInstance.startNetwork()` and `myInstance.stopNetwork()`; if one of these methods needs `this` but not the other, it doesn't make sense that one of the methods should be static or standalone: `MyClass.stopNetwork()`. Maybe the first method needs to read `this._config` but the stop method doesn't. 
- The usage of `this` within a method is an implementation detail and should not be a factor when deciding how a function should be exposed.

Reply:
> Abstract methods: Maybe the method just throws an error (so it doesn't use this); if it's designed to be overridden.
The method is not yet fully implemented; it doesn't use this right now, but it will soon, but we want to commit the code anyway.

Typescript wise abstract method shouldn't have implementation. Javascript wise, there is no abstract method. If it needs to be used in that way, we could say this is an exception, which shouldn't be the general case.

> One method which doesn't use this is functionally related to another method which does use this. For example myInstance.startNetwork() and myInstance.stopNetwork(); if one of these methods needs this but not the other, it doesn't make sense that one of the methods should be static or standalone: MyClass.stopNetwork(). Maybe the first method needs to read this._config but the stop method doesn't.

this is definitely a valid case against this, but I think we can put this as an exception. 

> The usage of this within a method is an implementation detail and should not be a factor when deciding how a function should be exposed. I don't think that we should let the linter tell us what is the right interface based on our implementation.

I think this is important to prevent meaning less state function, and prefer stateless functions.
Whether to expose function or not is not limited by this rule imo.